authorize/evaluator/headers_evaluator_evaluation.go

```diff
@@ -97,26 +103,42 @@ func (e *headersEvaluatorEvaluation) fillKubernetesHeaders(ctx context.Context)
 }
 
 func (e *headersEvaluatorEvaluation) fillGoogleCloudServerlessHeaders(ctx context.Context) {
-	if e.request.EnableGoogleCloudServerlessAuthentication {
-		h, err := getGoogleCloudServerlessHeaders(e.evaluator.store.GetGoogleCloudServerlessAuthenticationServiceAccount(), e.request.ToAudience)
-		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msg("authorize/header-evaluator: error retrieving google cloud serverless headers")
-			return
-		}
-		for k, v := range h {
-			e.response.Headers.Add(k, v)
-		}
+	if e.request.Policy == nil || !e.request.Policy.EnableGoogleCloudServerlessAuthentication {
+		return
+	}
+
+	var toAudience string
+	for _, wu := range e.request.Policy.To {
+		toAudience = "https://" + wu.URL.Hostname()
+	}
```

Comment:
this looks like it might have been incorrect in the original code, should it be building a list of audiences?